.github/workflows/ci.yml

```diff
@@ -0,0 +1,65 @@
+name: Quality Assurance
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build-linux:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+        platform: [x32, x64]
+        compiler: [gcc, clang]
+    steps:
+      - name: Setup Dependencies
+        run: |
+          sudo apt-get update -y -qq
+          sudo apt-get install -y gcc g++ make autoconf automake make libyajl-dev libxml2-dev libmaxminddb-dev libpcre2-dev libpcre2-8-0 libpcre2-16-0 libpcre2-32-0 libcurl4-gnutls-dev jq wget
+      - name: Install ModSecurity library
+        run: |
+          modsec=`curl -s https://api.github.com/repos/owasp-modsecurity/ModSecurity/releases/latest | jq .assets[0].browser_download_url | tr -d "\""`
+          modsecver=`curl -s https://api.github.com/repos/owasp-modsecurity/ModSecurity/releases/latest | jq .name | tr -d "\""`
+          wget $modsec -O modsecurity.tar.gz
+          tar -xzf modsecurity.tar.gz
+          cd modsecurity-$modsecver
+          ./configure --disable-lmdb --prefix=/usr
+          make -j `nproc`
+          sudo make install
+          cd
+      - uses: actions/checkout@v2
+        with:
+          path: ModSecurity-nginx
+      - name: Get Nginx source
+        uses: actions/checkout@v3
+        with:
+          repository: nginx/nginx
+          path: nginx
+      - name: Build nginx
+        working-directory: nginx
+        run: |
+          ./auto/configure --with-ld-opt="-Wl,-rpath,/usr/local/lib" --without-pcre2 --add-module=/home/runner/work/ModSecurity-nginx/ModSecurity-nginx/ModSecurity-nginx
+          make
+          make modules
+          sudo make install
+      - name: Start Nginx
+        run: |
+          sudo /usr/local/nginx/sbin/nginx -c /home/runner/work/ModSecurity-nginx/ModSecurity-nginx/ModSecurity-nginx/.github/nginx.conf
+      - name: Run attack test vhost 1
+        run: |
+          resp=`curl -I -X GET -H "Host: modsectest1" http://localhost/?q=attack | head -1 | cut -d$' ' -f2`
+          if [ $resp == "403" ]; then echo "OK"; else echo "FAIL"; exit 1; fi
+      - name: Run non-attack test vhost 1
+        run: |
+          resp=`curl -I -X GET -H "Host: modsectest1" http://localhost/?q=1 | head -1 | cut -d$' ' -f2`
```

Comment:
```suggestion
          set -ex -o pipefail
          status=$(curl -sSo /dev/null -w %{http_code} -I -X GET -H "Host: modsectest1" "http://localhost/?q=1")
```

Reply:
Done, see 9bad4b0ebd143d9583bce4e3590610ee53bc126c.

I didn't put the `set -ex -o piperfail`, because the aim here is to catch the failed request. An `exit 1` is enough here, I guess.

Reply:
That was a remnant of a previous edit, which I had forgotten to remove. I agree, you don't need it.